nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -551,11 +548,43 @@ def _get_metrics_calculator(self) -> MetricsCalculator:
 
 
 class ActivationAPoZRankPruner(ActivationPruner):
+    def _collector(self, buffer: List) -> Callable[[Module, Tensor, Tensor], None]:
+        assert len(buffer) == 0, 'Buffer pass to activation pruner collector is not empty.'
+        # The length of the buffer used in this pruner will always be 2.
+        # buffer[0] is the number of how many batches are counted in buffer[1].
+        # buffer[1] is a tensor and the size of buffer[1] is same as the activation,
+        # each element in buffer[1] means the sum of zero counts of the same activation position cross each batch.
+        buffer.append(0)
+
+        def collect_activation(_module: Module, _input: Tensor, output: Tensor):
```

Comment:
APOZ's _collector looks exactly the same with MeanRank? why not put the common functions into `ActivationPruner`?

Reply:
Yes, it is a good idea, I will merge these two functions

Reply:
fix it